src/cache.rs

```diff
@@ -0,0 +1,53 @@
+use anyhow::Error;
+use aws_sdk_cloudfront::{
+    model::{InvalidationBatch, Paths},
+    Client, RetryConfig,
+};
+use tokio::runtime::Runtime;
+use uuid::Uuid;
+
+/// create a CloudFront invalidation request for a list of path patterns.
+/// patterns can be
+/// * `/filename.ext` (a specific path)
+/// * `/directory-path/file-name.*` (delete these files, all extensions)
+/// * `/directory-path/*` (invalidate all of the files in a directory, without subdirectories)
+/// * `/directory-path*` (recursive directory delete, including subdirectories)
+/// see https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/Invalidation.html#invalidation-specifying-objects
+///
+/// Returns the caller reference that can be used to query the status of this
+/// invalidation request.
+pub(crate) fn create_cloudfront_invalidation(
```

Comment:
I tested this method manually with a cloudfront distribution on another account.